src/ngraph/frontend/onnx_import/utils/convpool.hpp

```diff
@@ -84,33 +87,55 @@ namespace ngraph
                 return get_pads(node, get_kernel_shape(node));
             }
 
-            /**
-             * @brief Create an nGraph pooling operation based on an ONNX pooling op.
-             *
-             * @tparam T Class of an nGraph pooling operation (e.g. AveragePool, MaxPool)
-             * @param node incoming ONNX opearation
-             * @return nGraph node equivalent of the ONNX operation
-             */
+            /// \ brief Create an nGraph pooling operation based on an ONNX pooling op.
+            ///
+            /// \ param T Class of an nGraph pooling operation (e.g. AveragePool, MaxPool)
+            /// \ param node incoming ONNX opearation
+            /// \ return nGraph node equivalent of the ONNX operation
             template <class T>
             inline NodeVector make_ng_pool(const Node& node)
             {
                 // Fetch input node for the pooling operation
                 auto data = node.get_ng_inputs().at(0);
 
                 // Parse ONNX op attributes
-                Shape kernel_shape = convpool::get_kernel_shape(node);
+                Shape kernel_shape;
+                if (node.op_type().find("Global") != std::string::npos)
+                {
+                    kernel_shape = node.get_ng_inputs()[0]->get_shape();
+                    kernel_shape.erase(std::begin(kernel_shape),
+                                       std::next(std::begin(kernel_shape), 2));
+                }
+                else
+                {
+                    kernel_shape = convpool::get_kernel_shape(node);
+                }
                 auto strides = convpool::get_strides(node);
                 auto dilations = convpool::get_dilations(node);
                 auto paddings = convpool::get_pads(node);
 
+                bool count_include_pad = node.get_attribute_value<int64_t>("count_include_pad", 0);
+
                 // Convert padding from CoordinateDiff to Shape objects
                 const CoordinateDiff& padding_above{paddings.first};
                 const CoordinateDiff& padding_below{paddings.second};
                 Shape padding_below_shape{std::begin(padding_below), std::end(padding_below)};
```

Comment:
I see that we use `get_pads` only here, so maybe we should consider returning `Shape` object instead of `CoordinateDiff` ?